File: 0001-Two-Sum/lc0001.js
// JavaScript solution.
/**
 * @param  {number[]} nums
 * @param  {number}   target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
    let hash = new Map();
    for (const [k, v] of nums.entries()) {
        if (hash.has(target-v)) return [hash.get(target-v), k];
        else hash.set(v, k);
    }
    return [];
};

File: 0001-Two-Sum/lc0001.go
// Go solution.

func twoSum(nums []int, target int) []int {
    hash := make(map[int]int)
    for idx, num := range nums {
        _, exist := hash[num]
        if exist {
            return []int{hash[num], idx}
        } else {
            hash[target-num] = idx
        }
    }
    return nil
}